tests/basic/pool_function_test.go

```diff
@@ -859,3 +860,87 @@ var _ = Describe("{PoolExpandAddDiskInMaintenanceMode}", func() {
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/tests/view/34542894
+
+	BeforeEach(func() {
+		StartTorpedoTest("PoolExpandAndCheckAlertsUsingResizeDisk", "pool expansion using resize-disk and check alerts after each operation", nil, testrailID)
+		contexts = scheduleApps()
+	})
+	JustBeforeEach(func() {
+		poolIDToResize = pickPoolToResize()
+		log.Infof("Picked pool %s to resize", poolIDToResize)
+		// poolToResize = getStoragePool(poolIDToResize)
+		storageNode, err = GetNodeWithGivenPoolID(poolIDToResize)
+		log.FailOnError(err, "Failed to get node with given pool ID")
+	})
+	JustAfterEach(func() {
+		AfterEachTest(contexts)
+	})
+
+	AfterEach(func() {
+		appsValidateAndDestroy(contexts)
+		EndTorpedoTest()
+	})
+
+	It("pool expansion using resize-disk and check alerts after each operation", func() {
+		log.InfoD("Initiate pool expansion using resize-disk")
+		poolToResize = getStoragePool(poolIDToResize)
+		originalSizeInBytes = poolToResize.TotalSize
+		targetSizeInBytes = originalSizeInBytes + 100*units.GiB
+		targetSizeGiB = targetSizeInBytes / units.GiB
+		log.InfoD("Current Size of the pool %s is %d GiB. Trying to expand to %v GiB with type resize-disk", poolIDToResize, poolToResize.TotalSize/units.GiB, targetSizeGiB)
+		triggerPoolExpansion(poolIDToResize, targetSizeGiB, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK)
+
+		resizeErr := waitForOngoingPoolExpansionToComplete(poolIDToResize)
+		dash.VerifyFatal(resizeErr, nil, "Pool expansion does not result in error")
+
+		log.Infof("Check the alert for pool expand for pool uuid %s", poolIDToResize)
+		alertExists, _ := checkAlertsForPoolExpansion(poolIDToResize, targetSizeGiB)
+		dash.VerifyFatal(alertExists, true, "Verify Alert is Present")
+	})
+
+})
+
+func checkAlertsForPoolExpansion(poolIDToResize string, targetSizeGiB uint64) (bool, error) {
+	// Get the node to check the pool show output
+	n := node.GetStorageDriverNodes()[0]
+	// Below command to change when PWX-28484 is fixed
+	cmd := "pxctl alerts show| grep -e POOL"
+	// Execute the command and check the alerts of type POOL
+	out, err := Inst().N.RunCommandWithNoRetry(n, cmd, node.ConnectionOpts{
+		Timeout:         2 * time.Minute,
+		TimeBeforeRetry: 10 * time.Second,
+	})
+	log.FailOnError(err, "Unable to execute the alerts show command")
+	outLines := strings.Split(out, "\n")
+	// var alertExist bool
+	// alertExist = false
+	substr := "[0-9]+ GiB"
+	re := regexp.MustCompile(substr)
+
+	for _, l := range outLines {
+		line := strings.Trim(l, " ")
+		if strings.Contains(line, "PoolExpandSuccessful") && strings.Contains(line, poolIDToResize) {
+			if re.MatchString(line) {
+				matchedSize := re.FindStringSubmatch(line)[0]
+				poolSize := matchedSize[:len(matchedSize)-4]
+				poolSizeUint, _ := strconv.ParseUint(poolSize, 10, 64)
+				if poolSizeUint >= targetSizeGiB {
+					// alertExist = true
+					log.Infof("The Alert generated is %s", line)
+					return true, nil
+					// break
+				} else {
```

Comment:
This else block is not required. 
If `PoolExpandSuccessful` alert is seen for given `poolIDToResize` and for `targetSizeGiB`, only then return true(ie. Alert found), In all rest of the cases, (ie. when for loop exits), just return false, error (line no. 945) that expected alert not found.

Reply:
fixed.